bundle/src/main/java/com/adobe/acs/commons/dam/impl/AssetsFolderPropertiesSupport.java

```diff
@@ -0,0 +1,206 @@
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.api.wrappers.CompositeValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component(
+        label = "ACS AEM Commons - Asset Folder Properties Support",
+        metatype = true,
+        policy = ConfigurationPolicy.REQUIRE
+)
+@Properties({
+        @Property(
+                name = "sling.filter.scope",
+                value = "request",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "filter.order",
+                intValue = 10000,
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.methods",
+                value = "GET",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.resourceTypes",
+                value = "acs-commons/touchui-widgets/asset-folder-properties-support",
+                propertyPrivate = true
+        ),
+})
+@Service
+public class AssetsFolderPropertiesSupport extends SlingSafeMethodsServlet implements Filter {
+    private static final Logger log = LoggerFactory.getLogger(AssetsFolderPropertiesSupport.class);
+
+    private static final String[] DEFAULT_PROPERTIES = new String[]{};
+    private List<String> properties = null;
+    @Property(label = "Allowed properties to persist",
+            description = "Relative paths from /content/dam/.../[sling:Folder] | [sling:OrderedFolder]",
+            value = {})
+    public static final String PROP_PROPERTIES = "properties";
+
+    public static final String GRANITE_UI_FORM_VALUES = "granite.ui.form.values";
+
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
+        final SlingHttpServletRequest request = (SlingHttpServletRequest) servletRequest;
+
+        if (!accepts(request)) {
+            chain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        chain.doFilter(servletRequest, servletResponse);
+
+        if (log.isTraceEnabled()) {
+            log.trace("Accepted by AssetsFolderPropertiesSupport [ {} ]", request.getResource().getPath());
+        }
+
+        for (final String property : properties) {
+            log.debug("Looking for request parameter named [ {} ] in [ {} ]", property, StringUtils.join(request.getRequestParameterMap().keySet()));
+            boolean deleteHint = false;
+            boolean propertyParamExists = false;
+
+            if (request.getRequestParameterMap().keySet().contains(property)) {
```

Comment:
I'd recommend grabbing keySet into a var, the number of nested calls and the fact you have to do it twice.